test/interval.jl

```diff
@@ -144,6 +143,141 @@
         @test span(interval) == Hour(3)
     end
 
+    @testset "maximum/minimum" begin
+        # Helper functions that manage the value we should be expecting from min and max.
+        function _min_val_helper(interval, a, unit)
+            t = eltype(interval)
+            # If the interal is empty, min is nothing
+            isempty(interval) && return nothing
+
+            # If a is in the interval, it is closed/unbounded and min is the first value.
+            # If a is nothing then it is unbounded and min is typemin(T)
+            a === nothing && return typemin(t)
+            a ∈ interval && return first(interval)
+
+            # From this point on, b ∉ interval so the bound is Open
+            # Also, if a is infinite we return typemin
+            # If it's an abstractfloat, we can't return just typemin since typemin IS Inf and
+            # since the bound is open at this point, Inf ∉ interval So we return the one after INF
+            !Intervals.isfinite(a) && t <: AbstractFloat && return nextfloat(a)
+            !Intervals.isfinite(a) && return typemin(t)
+
+            f = first(interval)
+            nv = if t <: AbstractFloat && unit === nothing
+                nextfloat(f)
+            else
+                f + unit
+            end
+
+            nv ∈ interval && return nv
+
+            # If we get to this point, the min/max functions throw a DomainError
+            # Since we want our tests to be predictable, we will not throw an error in this helper.
+        end
+
+        function _max_val_helper(interval, b, unit)
+            t = eltype(interval)
+            # If the interal is empty, min is nothing
+            isempty(interval) && return nothing
+
+            # If a is in the interval, it is closed/unbounded and min is the first value.
+            # If a is nothing then it is unbounded and min is typemin(T)
+            b === nothing && return typemax(t)
+            b ∈ interval && return last(interval)
+
+            # From this point on, b ∉ interval so the bound is Open
+            # Also, if a is infinite we return typemin
+            # If it's an abstractfloat, we can't return just typemin since typemin IS Inf and
+            # since the bound is open at this point, Inf ∉ interval So we return the one after INF
+            !isfinite(b) && t <: AbstractFloat && return prevfloat(b)
+            !isfinite(b) && return typemax(t)
+
+            l = last(interval)
+            nv = if t <: AbstractFloat && unit === nothing
+                prevfloat(l)
+            else
+                l - unit
+            end
+
+            nv ∈ interval && return nv
+
+            # If we get to this point, the min/max functions throw a DomainError
+            # Since we want our tests to be predictable, we will not throw an error in this helper.
+        end
```

Comment:
As theses helpers pretty much just re-implement minimum/maximum there aren't that helpful with testing. I think we can proceed with this but something to keep in mind for future work.

Reply:
I agree, in hindsight it would've been best to separate the tests per edge case and avoid having to do this.